### DISCUSSION TITLE
Is there an example of reading xdata data

### DISCUSSION THREAD

**Discussion by yingyue4008** [LibreDWG/libredwg#566] [Q&A]
I'm currently using the api to read xdata data, but I don't know dwg_obj_xrecord_get_xdata How to use the parameters of the xdata function? Please help me. Thank you very much

**Comment by yingyue4008** [comment]
At present, I have found the relevant logical method. I will try to solve it first. Thank you

**Comment by yingyue4008** [comment]


**Reply by rurban** [reply]
If it crashes you need to check why

**Comment by rurban** [comment]
You need to check the dwg version if it's a char* or a Unicode string. The name API functions convert that to utf8 which is more convenient.

**Reply by yingyue4008** [reply]
Thank you. I have solved this problem through your prompt

**Reply by yingyue4008** [reply]
Hello, I have found that I can get the value of 1000 group codes, but I don't know how to get the value of 1070 group codes. I hope to get some help
